fe/src/main/java/org/apache/doris/analysis/CaseExpr.java

```diff
@@ -186,6 +189,9 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
             // Determine maximum compatible type of the then exprs seen so far.
             // We will add casts to them at the very end.
             Expr thenExpr = children.get(i + 1);
+            if (!thenExpr.getType().isScalarType()) {
+                throw new AnalysisException("subquery in case-when must return scala type");
+            }
```

Comment:
why do you only thow exception where expr is ‘when expr’ and ‘then expr’？
Whether to throw an exception when ‘case expr’ and ‘else expr’ is not scarlar type？

Reply:
if `when exp`or `case expr` has subquery and subquer returns non-scalar type, the error will be thrown by `analyzer.getCompatibleType` and the message is  weird, too difficult to understand, so i throw a message here